### PR TITLE
NCC in 2-net-a-simple

### PR DESCRIPTION
**Pull request by sruffilli**
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass
- [x] Updated README.md to reflect the changes

This is a first, basic implementation of NCC in 2-simple.

Caveats: 

1. Landing is not part of the core network anymore, but it simply hosts hybrid connectivity, which itself is connected as a spoke
2. Because of (1), do we need a "shared services" VPC?
3. VPC spokes must be authorized manually (i.e. from the Cloud Console). At this stage the terraform provider (nor the API?) does not allow for ncc hub projects allowlisting.

### CODE REVIEW

File: fast/stages/2-networking-a-simple/main.tf [hunk 1]
```diff
@@ -22,7 +22,11 @@ locals {
     for k, v in coalesce(var.service_accounts, {}) :
     k => "serviceAccount:${v}" if v != null
   }
-  spoke_connection = var.spoke_configs.peering_configs != null ? "peering" : "vpn"
+  spoke_connection = coalesce(
```
Comment (LucaPrete) [inline fast/stages/2-networking-a-simple/main.tf:25]: I'm wondering whether we should delete this and directly use the variables in the code. It seems it doesn't really help reducing the complexity

File: fast/stages/2-networking-a-simple/spoke-ncc.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2022 Google LLC
```
Comment (LucaPrete) [inline fast/stages/2-networking-a-simple/spoke-ncc.tf:2]: nit: I'd take the chance to update the copyrights of all files we touch

**Review by LucaPrete — COMMENTED** [review]
Minor comments.
For ncc we shouldn't need to be whitelisted, given it just went public preview. Anyway, the provider is not there yet.
Also, readme should be updated before merge.

File: fast/stages/2-networking-a-simple/main.tf (hunk 1 above)
Comment (ludoo) [reply fast/stages/2-networking-a-simple/main.tf:25]: uh? I think this is a good way of doing it...

File: fast/stages/2-networking-a-simple/variables.tf [hunk 3]
```diff
@@ -130,6 +130,18 @@ variable "regions" {
 variable "spoke_configs" {
   description = "Spoke connectivity configurations."
   type = object({
+    ncc_configs = optional(object({
+      dev = optional(object({
```
Comment (ludoo) [inline fast/stages/2-networking-a-simple/variables.tf:134]: nit: I would add a linebreak after the comma for legibility

**Review by ludoo — COMMENTED** [review]
really linear and simple to read :)

**Comment by sruffilli** [comment]
> Minor comments. For ncc we shouldn't need to be whitelisted, given it just went public preview. Anyway, the provider is not there yet. 

This PR is tested and it works :) 
The provider didn't really require any change, as the new feature allows for the NCC-Hub to have BOTH VPC spokes and Hybrid connectivity spokes.

> Also, readme should be updated before merge.

This is why the PR is in draft. I'm updating the README as/if we reach consensus this is a valid PR.

File: fast/stages/2-networking-a-simple/spoke-ncc.tf (hunk 2 above)
Comment (sruffilli) [reply fast/stages/2-networking-a-simple/spoke-ncc.tf:2]: I already had another commit lined up, but didn't want to ruin the party for you. 
Coming in :)

File: fast/stages/2-networking-a-simple/main.tf (hunk 1 above)
Comment (sruffilli) [reply fast/stages/2-networking-a-simple/main.tf:25]: I thought a bit about a potentially more elegant solution but I didn't find any. 
At some point you have an N-nary decision to make.

Comment (LucaPrete) [reply fast/stages/2-networking-a-simple/main.tf:25]: OK, understood. lgtm. Feel free to resolve the comment.

**Review by LucaPrete — COMMENTED** [review]
lgtm now. I guess we can finalize this updating tests and readme and then merge.

**Comment by sruffilli** [comment]
Given there seems to be consensus, I'll go ahead and write docs/tests.

**Review by ludoo — APPROVED** [review]

**Review by juliocc — APPROVED** [review]

**Review by LucaPrete — APPROVED** [review]

**Pull request merged**